src/Core.UnitTests/CFamily/EffectiveRulesConfigCalculatorTests.cs

```diff
@@ -56,7 +61,33 @@ public void GetConfig_RulesInCustomSettings_MergedConfigReturned()
             var testLogger = new TestLogger();
             var testSubject = new EffectiveRulesConfigCalculator(testLogger);
 
-            var defaultRulesConfig = CreateMockConfig("key");
+            var defaultRulesConfig = CreateWellKnownRulesConfig("key");
+            var sourcesSettings = new RulesSettings
+            {
+                Rules = new Dictionary<string, RuleConfig>
+                {
+                    // Turn an active rule off...
+                    { "key:" + WellKnownPartialRuleKey1_Active, new RuleConfig { Level = RuleLevel.Off } },
+                    // ... and an inactive rule one
+                    { "key:" + WellKnownPartialRuleKey3_Inactive, new RuleConfig { Level = RuleLevel.On } }
+                }
+            };
+
+            var result = testSubject.GetEffectiveRulesConfig("language1", defaultRulesConfig, sourcesSettings);
+
+            result.LanguageKey.Should().Be("key");
+            result.AllPartialRuleKeys.Should().BeEquivalentTo(defaultRulesConfig.AllPartialRuleKeys);
+            result.ActivePartialRuleKeys.Should().BeEquivalentTo(WellKnownPartialRuleKey2_Active, WellKnownPartialRuleKey3_Inactive);
+        }
+
+        [TestMethod]
+        public void GetConfig_CachedResultsReturnedIfAvailable()
```

Comment:
I don't think that we should rely on the logger as an indication of cache.
A better way IMO would be to change the sourceSettings on the second call, and to see that the changes are not being taken into effect.
I'd also remove assertions that have already been made in other tests, such as testing the results of call #1 - it was already tested in 2 tests above.

Reply:
The test is checking the object identity to see if the cache is working.
The logger is being checked to see that the notifications are correct.